modules/ycc_vm.py

```diff
@@ -510,7 +510,7 @@ def _compare_disk(self, disk_id, disk_spec):
         if str(disk_spec["size"]) != disk["size"]:
             err.append("size")
 
-        if disk_spec.get("image_id") and disk_spec["image_id"] != disk["sourceImageId"]:
+        if disk_spec.get("image_id") and disk.get("sourceImageId") and disk_spec["image_id"] != disk["sourceImageId"]:
```

Comment:
Should it be `if disk_spec.get("image_id") and (not disk.get("sourceImageId") or disk_spec["image_id"] != disk["sourceImageId"]):`?